### ISSUE TITLE
Possible memory leak (different errors show up after iterating hundreds of simulations automatedly) 

### ISSUE THREAD

**Issue by lguilhermers** [architecture-building-systems/CityEnergyAnalyst#3067]
**Describe the bug**
This bugs relates to the stochastic scenario generator (SSG) developed in #2998 .
The SSG performs multiple iterations of the same scenario (while changing only some of the input data) and stores some inputs and outputs of each simulation.
I noticed that every time I reach iteration 405 (sometimes 404 or 406), I get different errors related to the simulation. This happened for around 10 tries, so I know it is very unlikely that is indeed a simulation error, end most likely a memory issue (@reyery also agrees that should be a memory leak somewhere).
When running the code in parallel (which means one scenario for each core),  the error is given once one of the cores reach its own iteration number 405.
PS: I have tested only on my laptop. Not sure if this threshold would be the same across different machines.

**To Reproduce**
1. Run stochastic scenario generator code (you have to ask directly to me as it hasn't been deployed as plugin)
2. Wait until iteration 404-406 (might be different in other machines)
3. See diverse errors (see below for some examples)

**Comment by lguilhermers** [comment]
Here are some of the errors that I got:

```
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "C:\Users\luis.santos\Documents\CityEnergyAnalyst\Dependencies\Python\lib\multiprocessing\pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\luis.santos\Documents\CityEnergyAnalyst\Dependencies\Python\lib\multiprocessing\pool.py", line 44, in mapstar
    return list(map(*args))
  File "c:\users\luis.santos\documents\cityenergyanalyst\cityenergyanalyst\cea\utilities\parallel.py", line 131, in __apply_func_with_worker_stream
    result = func(*args)
  File "C:\Users\luis.santos\Documents\GitHub\stochastic_scenario_generator\data_generation.py", line 164, in stochastic_scenario_generator
    demand_main.demand_calculation(locator, config)  # runs demand simulation
  File "c:\users\luis.santos\documents\cityenergyanalyst\cityenergyanalyst\cea\demand\demand_main.py", line 130, in demand_calculation
    repeat(debug, n))
  File "c:\users\luis.santos\documents\cityenergyanalyst\cityenergyanalyst\cea\utilities\parallel.py", line 149, in wrapper
    result = func(*instance_args)
  File "c:\users\luis.santos\documents\cityenergyanalyst\cityenergyanalyst\cea\demand\thermal_loads.py", line 140, in calc_thermal_loads
    tsd = electrical_loads.calc_Eaux_fw(tsd, bpr, schedules)
  File "c:\users\luis.santos\documents\cityenergyanalyst\cityenergyanalyst\cea\demand\electrical_loads.py", line 143, in calc_Eaux_fw
    tsd['Eaux_fw'] = np.vectorize(calc_Eauxf_fw)(tsd['vfw_m3perh'], deltaP_kPa, b)
  File "C:\Users\luis.santos\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\numpy\lib\function_base.py", line 2108, in __call__
    return self._vectorize_call(func=func, args=vargs)
  File "C:\Users\luis.santos\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\numpy\lib\function_base.py", line 2195, in _vectorize_call
    res = array(outputs, copy=False, subok=True, dtype=otypes[0])
TypeError: can't convert complex to float
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:/Users/luis.santos/Documents/GitHub/stochastic_scenario_generator/data_generation.py", line 323, in <module>
    main(cea.config.Configuration())
  File "C:/Users/luis.santos/Documents/GitHub/stochastic_scenario_generator/data_generation.py", line 243, in main
    list_of_dataframe_with_instances)
  File "c:\users\luis.santos\documents\cityenergyanalyst\cityenergyanalyst\cea\utilities\parallel.py", line 96, in wrapper
    result = map_result.get()
  File "C:\Users\luis.santos\Documents\CityEnergyAnalyst\Dependencies\Python\lib\multiprocessing\pool.py", line 657, in get
    raise self._value
TypeError: can't convert complex to float

Process finished with exit code 1

```

**Comment by shanshanhsieh** [comment]
@MatNif will help @lguilhermers to reproduce this error.

**Comment by lguilhermers** [comment]
@MatNif thanks for the help! Can you try downloading from the following link and run the code for the scenario also attached?
https://we.tl/t-Ek6NmFlXTP

What happens to me is that always after 405-410 simulation I get diverse errors. I think is not related to the scenario itself, since it runs hundreds of satisfactory simulations. It would be great to know if the same happens in other machines.

**Comment by lguilhermers** [comment]
@MatNif  and I ran a few more cases and it is a bit unclear what raises the error, but I will share some findings:

- For a high number of iterations (2000+) of #2998 plugin, we got errors every time we ran the simulation.
- Some of those error messages that I showed above also show when running only this scenario using the GUI. This indicates (probably) that this combination of parameters is problematic for CEA. I don't think a single parameter is responsible for it though, as it wasn't a maximum or minimum value compared to other successful iterations.
- The script runs in parallel different scenarios. When one has an error, the others continue to be simulated until an error messages stops all simulations. This error message is always the same as the scenario that got the error first (but doesn't show up until much later it happens). Could this be related to how the parallelization  is raising errors?
- For me the script always stops around 400-405 iterations. Mathias managed to get different values, some of them even completing one of the scenarios (still raising an error in the script though). This suggests a different behavior between Windows and Mac (memory related?).

**TL,DR: the errors (or most of them) that show up in the script are reproduceable within specific scenarios. I suggest opening different issues for each one, uploading the problematic scenario and having a hackathon session to debug them.**
